### PR TITLE
Added a generic config for the JFrog Artifactory REST API

### PR DESCRIPTION
**Pull request by mdodell**
### What does this PR do?
- Added a generic connector config for the JFrog Artifactory Rest API

### What ticket does this PR close?
Connected to #1263 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] This PR does not require updating any documentation, or
- [x] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs

#### (For releases only) Manual tests
- [ ] Manually tested [Keychain provider](https://github.com/cyberark/secretless-broker/tree/master/test/providers/keychain)
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local Secretless Broker image build of your branch

### CODE REVIEW

**Comment by izgeri** [comment]
this one would be a cool demo too :)

**Comment by mdodell** [comment]
> this one would be a cool demo too :)

I think this work flow could be a very simple demo that showcases it. I saw [this example for deploying an Artifact](https://www.jfrog.com/confluence/display/RTF6X/Artifactory+REST+API) on their docs. It would be cool to show how it changes before/after using Secretless.

File: examples/generic_connector_configs/README.md [hunk 1]
```diff
@@ -353,6 +358,85 @@ from the correct provider/path.
...
+The configuration file for the GitHub API can be found at
+[jfrog_artifactory_secretless.yml](./jfrog_artifactory_secretless.yml).
+
+> This configuration uses
+[v6.x](https://www.jfrog.com/confluence/display/RTF6X/Artifactory+REST+API)
+of the JFrog Artifactory REST Api.
+
+#### How to use this connector
+* Edit the supplied configuration to get your
+JFrog Artifactory API Key[https://www.jfrog.com/confluence/display/JFROG/User+Profile#UserProfile-APIKey],
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:375]:
```suggestion
[JFrog Artifactory API Key](https://www.jfrog.com/confluence/display/JFROG/User+Profile#UserProfile-APIKey),
```

File: examples/generic_connector_configs/README.md [hunk 2]
```diff
@@ -353,6 +358,85 @@ from the correct provider/path.
...
+JFrog Artifactory Access Token[https://www.jfrog.com/confluence/display/JFROG/Access+Tokens]
```
Comment (JakeQuilty) [inline examples/generic_connector_configs/README.md:376]:
```suggestion
[JFrog Artifactory Access Token](https://www.jfrog.com/confluence/display/JFROG/Access+Tokens)
```

**Pull request merged**